fs/xfstests.py

```diff
@@ -221,6 +229,73 @@ def setUp(self):
+                build.make(e2fsprogs_build_dir, extra_args='install')
+
+            if self.fs_to_test == "xfs":
+                if self.detected_distro.name in ['centos', 'fedora', 'rhel']:
+                    libini_path = process.run("ldconfig -p | grep libini", verbose=True, ignore_status=True)
+                    if not libini_path:
+                        # Build libini.h as it is needed for xfsprogs
+                        libini_dir = os.path.join(self.teststmpdir, 'libini')
+                        if not os.path.exists(libini_dir):
+                            os.makedirs(libini_dir)
+                        git.get_repo('https://github.com/benhoyt/inih',
+                                     destination_dir=libini_dir)
+                        os.chdir(libini_dir)
+                        process.run("meson build", verbose=True)
+                        libini_build_dir = os.path.join(libini_dir, 'build')
+                        if os.path.exists(libini_build_dir):
+                            os.chdir(libini_build_dir)
+                            process.run("meson install", verbose=True)
+                        else:
+                            self.fail('Something went wrong while building libini. \
+                                       Please check the logs.')
+                # Build xfs progs
+                xfsprogs_dir = os.path.join(self.teststmpdir, 'xfsprogs')
+                if not os.path.exists(xfsprogs_dir):
+                    os.makedirs(xfsprogs_dir)
+                xfsprogs_url = self.params.get('xfsprogs_url')
+                git.get_repo(xfsprogs_url, destination_dir=xfsprogs_dir)
+                os.chdir(xfsprogs_dir)
+                build.make(xfsprogs_dir)
+                build.make(xfsprogs_dir, extra_args='install')
+
+            if self.fs_to_test == "btrfs":
+                # Build btrfs progs
+                btrfsprogs_dir = os.path.join(self.teststmpdir, 'btrfsprogs')
+                if not os.path.exists(btrfsprogs_dir):
+                    os.makedirs(btrfsprogs_dir)
+                btrfsprogs_url = self.params.get('btrfsprogs_url')
+                git.get_repo(btrfsprogs_url, destination_dir=btrfsprogs_dir)
+                os.chdir(btrfsprogs_dir)
+                process.run("./autogen.sh", verbose=True)
+                process.run("./configure --prefix=/usr/local/ --disable-documentation", verbose=True)
+                build.make(btrfsprogs_dir)
+                build.make(btrfsprogs_dir, extra_args='install')
+
+        # Check versions of fsprogs
+        e2fs_ver = process.system_output("mkfs.ext4 -V", ignore_status=True, shell=True).decode("utf-8")
+        self.log.info(e2fs_ver)
+        xfs_ver = process.system_output("mkfs.xfs -V", ignore_status=True, shell=True).decode("utf-8")
+        self.log.info(xfs_ver)
+        btrfs_ver = process.system_output("mkfs.btrfs -V", ignore_status=True, shell=True).decode("utf-8")
+        self.log.info(btrfs_ver)
```

Comment:
Can you use self.fs_to_test and make the code generic (instead of repetitive) ? 
```
fs_ver = process.system_output("mkfs.%s -V" % self.fs_to_test, ignore_status=True, shell=True).decode("utf-8")
self.log.info(fs_ver)
```

Reply:
In fact the same logic can be applied to the above code block used to compile the <fs>progs code.

Reply:
Yes that sounds better Sachin, I've addressed this.

Thanks